src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -23,101 +23,196 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
 
     /**
      * Return road accident with matching index
+     *
      * @param index
      * @return
      */
-    public RoadAccident getAccidentByIndex7(String index){
+    public RoadAccident getAccidentByIndex7(String index) {
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+        while (roadAccidentIterator.hasNext()) {
```

Comment:
what about for(accident: list)? In this case don't need to create iterator. You can use such syntax with anything implementing Iteratable interface.